src/trs/lexicon.rs

```diff
@@ -805,7 +838,46 @@ impl Lex {
         }
         context.to_rule().or(Err(SampleError::Subterm))
     }
+    fn infer_term_inter(
```

Comment:
Can you please add some documentation? I'm not entirely sure what this function does?